data-prepper-api/src/main/java/org/opensearch/dataprepper/model/encryption/EncryptionHttpHandler.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.encryption;
+
+import com.sun.net.httpserver.HttpHandler;
```

Comment:
Let's find a way to keep this out of data-prepper-api.

A simple solution would be to create a coupling between data-prepper-core and extension-plugin. I recommend this over exposing this in data-prepper-api. Longer-term we can have a better approach for extending the server.